### PR TITLE
Hubbard U+V

### PR DESCRIPTION
**Pull request by mtaillefumier**
- Implemented the U+V term for colinear case
- Add missing reduction over k-points when calculating non-local occupation numbers
- Symmetrize the local occupation numbers
- Add yet another test case for NiO (with the 12 bonds for the +V term)
- Reorganized the calculations of the occupancies deriatives (hopefully it is clearer now)

- TODO : consider the full set of orboitals when orthogonalizing the hubbard wave functions.
- TODO : add fortran interface and extensive tests with QE

- the diff is long because of formatting.

### CODE REVIEW

File: src/geometry/force.cpp [hunk 1]
```diff
@@ -675,14 +675,47 @@ void Force::hubbard_force_add_k_contribution_collinear(K_point& kp__, Q_operator
...
+          auto Tr = nl.T();
+
+          vector3d<double> Ttot;
+          Ttot[0] = Tr[0];
+          Ttot[1] = Tr[1];
+          Ttot[2] = Tr[2];
+          assert(ib == 2 * il + 1);
+          assert(jb == 2 * jl + 1);
+          if (ia1 == ia) {
+            auto z1 = std::exp(double_complex(0, -twopi * dot(Ttot, kp__.vk())));
```
Comment (toxa81) [inline src/geometry/force.cpp:706]: dot() works on 3d-vectors of different type, so  `vector3d<double> Ttot` is not needed

File: src/hamiltonian/non_local_operator.hpp [hunk 2]
```diff
@@ -148,8 +151,38 @@ class U_operator
...
+          auto Tr = nl.T();
+          vector3d<double> Ttot;
+
+          Ttot[0] = Tr[0];
+          Ttot[1] = Tr[1];
+          Ttot[2] = Tr[2];
+          assert(ib == 2 * il + 1);
+          assert(jb == 2 * jl + 1);
+
+          auto z1 = std::exp(double_complex(0, twopi * dot(Ttot, vk_)));
```
Comment (toxa81) [inline src/hamiltonian/non_local_operator.hpp:175]: same here, dot() can take vector3d of different types

File: src/hamiltonian/non_local_operator.hpp [hunk 3]
```diff
@@ -164,6 +197,9 @@ class U_operator
         return offset_[ia__];
     }
 
+  vector3d<T> &vk() {
```
Comment (toxa81) [inline src/hamiltonian/non_local_operator.hpp:200]: formatting
```
vector3d<T> vk()
{
    return vk_;
}
```

**Review by toxa81 — CHANGES_REQUESTED** [review]
Just a few minor corrections are required

**Comment by mtaillefumier** [comment]
I am going to close this PR since I rewrote it from scratch

**Pull request closed without merging**